config/src/errors.rs

```diff
@@ -0,0 +1,180 @@
+//! Errors for graph-sdk crate
+//!
+
+use thiserror::Error;
+
+pub type DsnpGraphResult<T> = std::result::Result<T, DsnpGraphError>;
+
+#[derive(Debug, Error)]
+pub enum DsnpGraphError {
```

Comment:
q: why are all values `String` instead of, for instance, `DsnpUserId` or other types? Is it specifically for C-compatibility? It seems like the `thiserror` crate gives us a lot of flexibility to keep the underlying data types while still producing a `Display` trait with a formatted string.